### DISCUSSION TITLE
Contributing to HIP Discussions

### DISCUSSION THREAD

**Discussion by fredemmott** [facebook/hhvm#8786] [HIPs (Hack Improvement Process)]
# Background

Almost all languages have a formal process for changing the language and/or standard library. This includes older languages like C++, Java and Python, as well as more recent languages such as Kotlin, Swift and Rust. The reason these processes are put into place is because language design is hard. It is necessary to think about interactions with current features as well as future features, alternatives need to be weighed and considered and in some cases new areas of research need to be developed or applied. Language design also has a long time line. Major language changes can easily take more than a year to develop, so it is important to track the history of the proposal. Finally, given the cost of adding new features there needs to be a clear, agreed upon motivation for the change.

Historically we have not instilled a formal proposal process for Hack, since we wanted to keep the process light weight. However we are now finding that the lack of formal process is making it difficult to make and track our decisions.

# Creating HIPs

Send a pull request adding a HIP to `hphp/hack/doc/HIPs` using the template which will be committed shortly. The template should be fairly completely filled - compare to existing HIPs. For less fleshed-out ideas, a 'feature request' issue may be more appropriate.

# Providing Feedback

- Feel free to comment on pull requests
- We are experimenting with GitHub Discussions for public discussion; currently, all committed HIPs should have a discussion thread.

We currently aim to keep one discussion per HIP; please do not create additional discussions in the HIPs category.

**Comment by ashred420** [comment]
Sino pwede kontakin may idea ako jan

**Reply by lipuparida12312-png** [reply]
Lipu

**Reply by lipuparida12312-png** [reply]
Lipu
